app/controllers/dashboard/visit_groups_controller.rb

```diff
@@ -36,14 +36,19 @@ def create
     @service_request = ServiceRequest.find(params[:service_request_id])
     @sub_service_request = SubServiceRequest.find(params[:sub_service_request_id])
     @arm =  Arm.find(params[:visit_group][:arm_id])
-    if @arm.add_visit(params[:visit_group][:position], params[:visit_group][:day], params[:visit_group][:window_before], params[:visit_group][:window_after], params[:visit_group][:name], 'true')
-      @arm.increment!(:minimum_visit_count)
-      @service_request.relevant_service_providers_and_super_users.each do |identity|
-        create_visit_change_toast(identity, @sub_service_request) unless identity == @user
+    @visit_group = VisitGroup.new(params[:visit_group])
```

Comment:
What's the purpose of this extraction? It seems like add_visit is already doing some validations which should fail.